clang/test/SemaSYCL/check-work-group-size-hint-device.cpp

```diff
@@ -97,14 +101,14 @@ class Functor16x2x1 {
 
 class Functor4x4x4 {
 public:
-  [[sycl::work_group_size_hint(4, 4, 4)]] void operator()() const {};
+  [[sycl::work_group_size_hint(4, 4, 4)]] void operator()() const {}; // expected-warning {{'work_group_size_hint' attribute can only be applied to a SYCL kernel function}}
```

Comment:
Isn't this incorrect? This is the operator of the kernel object right? This should not be generating a warning right?

Reply:
It's functor however it issues diagnostic because it's not used as kernel in this test. It was not submitted to queue.